### DISCUSSION TITLE
How to use -- trust config module correctly?

### DISCUSSION THREAD

**Discussion by zhanchunhui** [ablab/spades#1282] [Q&A] [answered]
I have a question about the SPAdes software. I already have a config.fasta file (only this file) and I want to assembling contigs into scaffolds files (get the file of scaffolds.fasta). Is this possible? I try the -- trust config module. But it reported an error when I run it. The form of the error is as follows.(See pictures for details)
== Error == You should specify at least one unpaired, paired-end, or high-quality mate-pairs library!
Does anyone know how to solve this problem? 
![1714058252197](https://github.com/ablab/spades/assets/168107490/9bf8bbcb-7bbc-4fbf-9d2c-84e6fea73d82)

**Comment by asl** [comment, marked as answer]
> I have a question about the SPAdes software. I already have a config.fasta file (only this file) and I want to assembling contigs into scaffolds files (get the file of scaffolds.fasta). Is this possible? I

Well, you cannot do this without additional data as there is no information in contigs that would allow this. Also, trusted contigs is used for something else. From the manual:
```
In case you have contigs of the same genome generated by other assembler(s) and you wish to merge them into SPAdes assembly, you can specify additional contigs using --trusted-contigs or --untrusted-contigs. First option is used when high quality contigs are available. These contigs will be used for graph construction, gap closure and repeat resolution. Second option is used for less reliable contigs that may have more errors or contigs of unknown quality. These contigs will be used only for gap closure and repeat resolution. The number of additional contigs is unlimited.
```
and
```
To run SPAdes 3.15.5 you need at least one library of the following types:

 - Illumina paired-end/high-quality mate-pairs/unpaired reads
 - IonTorrent paired-end/high-quality mate-pairs/unpaired reads
 - PacBio CCS reads
```

So, long story short: you cannot do anything with just contigs without paired-end reads.

**Reply by zhanchunhui** [reply]
Ok, I got it. Thank you for your reply.
